subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/internal/dependencies/ModuleProjectConfigurationDependenciesNode.kt

```diff
@@ -17,15 +17,26 @@ internal class ModuleProjectConfigurationDependenciesNode(
     }
 
     fun allDependencies(): Set<Project> {
-        return dependencies.mapValues { (_, nodeSet) ->
-            nodeSet.flatMap { node ->
-                mutableSetOf(node.project).also {
-                    it.addAll(
-                        node.allDependencies()
-                    )
-                }
+        val dependencies = mutableSetOf<Project>()
```

Comment:
It helps to avoid recursive visits only for one call `allDependencies`
But we call it for each root app, so we could reduce complexity more if we will use global cache where the key is node and value is all it dependencies

Reply:
I've decided to keep it as it is.

Caching results between different apps is a controversial solution. 
It requires storing all transitive dependencies for each module. This is a lot of duplicated data that requires much more memory.

Traversing on direct dependencies took 2-4ms for an app. This is definitely not a bottleneck.